File: experiment/functions_local/local.js
function strPresent(str){
  return str != null && str.length > 0
}

function validateParams(params){
  var firstNameValid = strPresent(params.firstName);
  var lastNameValid = strPresent(params.lastName);
  var emailValid = strPresent(params.email) && params.email.indexOf('@') > -1;
  var postcodeValid = strPresent(params.postcode);

  return firstNameValid && lastNameValid && emailValid && postcodeValid
}

var params = {
  firstName: 'Jane',
  lastName: 'Doe',
  email: 'jane@doe.com',
  postcode: ''
}

result = validateParams(params);

File: experiment/functions_local/local.rb
def str_present(str)
  str && str.length > 0
end

def validate_params(params)
  first_name_valid = str_present params[:first_name]
  last_name_valid = str_present params[:last_name]
  email_valid = str_present(params[:email]) && params[:email].include?('@')
  postcode_valid = str_present params[:postcode]

  first_name_valid && last_name_valid && email_valid && postcode_valid
end

params = {
  first_name: 'Jane',
  last_name: 'Doe',
  email: 'jane@doe.com',
  postcode: ''
}

result = validate_params params
